File: TGME/src/main/java/tmge/model/GameStats.java
package tmge.model;

import java.time.LocalDateTime;

public class GameStats {
    private String playerName;
    private String gameName;
    private int score;
    private LocalDateTime timestamp;
    private int duration;

    public GameStats(String playerName, String gameName) {
        this.playerName = playerName;
        this.gameName = gameName;
        this.score = 0;
        this.timestamp = LocalDateTime.now();
        this.duration = 0;
    }

    public String getPlayerName() {
        return playerName;
    }

    public String getGameName() {
        return gameName;
    }

    public int getScore() {
        return score;
    }

    public LocalDateTime getTimestamp() {
        return timestamp;
    }

    public void updateScore(int result) {
        this.score = result;
        this.timestamp = LocalDateTime.now();
    }

    public void setDuration(int seconds) {
        this.duration = seconds;
    }

    public int getDuration() {
        return duration;
    }

    @Override
    public String toString() {
        return String.format(
                "Game: %s, Player: %s, Score: %d, Played on: %s",
                gameName, playerName, score, timestamp
        );
    }
}

File: TMGE/user/game_stats.py
from datetime import datetime

class GameStats:
    def __init__(self, game_name: str, score: int):
        self.game_name = game_name
        self.score = score
        self.timestamp = datetime.now()
        self.duration = 0
    
    def get_game_name(self) -> str:
        return self.game_name
    
    def get_score(self) -> int:
        return self.score
    
    def get_timestamp(self) -> datetime:
        return self.timestamp
